WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -17,10 +17,14 @@ protocol PointOfSaleAggregateModelProtocol {
     var cardReaderConnectionStatus: CardPresentPaymentReaderConnectionStatus { get }
     func connectCardReader()
     func disconnectCardReader()
+    var paymentState: PointOfSalePaymentState { get }
+    var cardPresentPaymentAlertViewModel: PointOfSaleCardPresentPaymentAlertType? { get set }
+    var cardPresentPaymentInlineMessage: PointOfSaleCardPresentPaymentMessageType? { get }
 
     @available(*, deprecated, message: "`allItems` is due for removal, use `itemListState` instead.")
     var allItems: [POSItem] { get }
     var itemListState: ItemListState { get }
+    var blockReturnToItemSelection: Bool { get }
```

Comment:
nit;
I am conflicted with the existence of such a variable within the aggregate model. It's only used within a `CartView` and determines the state of the back button.

Maybe it's one of those places where a local view model makes sense, `CartViewModel isAddButtonDisabled(_ paymentState: PointOfSalePaymentState)`

Reply:
Agreed, this should be in the view or a view model, I thought it was relied on elsewhere but it definitely isn't.

Here are the two options I've come up with:
1. 1afc358 – just move the logic to the view.
2. 5a43088 – move it to a local view model but pass the whole posModel
3. 6a545ec – only pass the paymentState and orderState

The view model already has the posModel, but I agree that it shouldn't.

I prefer 1 and 3. Probably 1 by a little, but the tests and code in 3 aren't too bad, and I'll give it a try with multi-parameter tests so we can check every combination.

WDYT? I can switch to any of these options.

Reply:
@staskus I'm going to merge with number 3, but happy to change in a future PR if we decide on one of the other options.